syscall_darwin.go

```diff
@@ -7,32 +7,82 @@ import (
 	"unsafe"
 )
 
-func getxattr(path string, name string, value *byte, size int, pos int, options int) (int, error) {
-	r0, _, e1 := syscall.Syscall6(syscall.SYS_GETXATTR, uintptr(unsafe.Pointer(syscall.StringBytePtr(path))), uintptr(unsafe.Pointer(syscall.StringBytePtr(name))), uintptr(unsafe.Pointer(value)), uintptr(size), uintptr(pos), uintptr(options))
-	if e1 != syscall.Errno(0) {
-		return int(r0), e1
+func getxattr(path string, attr string, dest []byte) (sz int, err error) {
+	size := len(dest)
+	var value *byte
+	if size > 0 {
+		value = &dest[0]
 	}
-	return int(r0), nil
+	return darwinGetxattr(path, attr, value, size, 0, 0)
```

Comment:
we are already in `darwin`, so `darwin` prefix looks weired here. Moreover following code:
```go
size := len(dest)
var value *byte
if size > 0 {
    value = &dest[0]
 }
```
repeats in many places, so it does not look like a dedup.